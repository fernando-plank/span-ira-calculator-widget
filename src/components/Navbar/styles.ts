import styled from 'styled-components'

export const Wrapper = styled.div`
  width: 100%;
  height: 64px;
  background: ${({ theme }) => theme.colors.white};
  display: flex;
  align-items: center;
  justify-content: space-between;
`

export const MenuIconWrapper = styled.div`
  width: 80px;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
`

export const ExplorePanelWrapper = styled.div`
  heigth: 24px;
  margin-right: 20px;
  cursor: pointer;
`
