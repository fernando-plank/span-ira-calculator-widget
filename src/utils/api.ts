import { IncentivesResponse, TaxCreditsIncentives } from '@services/types'
import { makeTaxCreditsTooltip } from './tooltips'

export const formatTaxCreditsIncentives = (
  tax_credit_incentives: TaxCreditsIncentives[]
): TaxCreditsIncentives[] => {
  let eletricPanel: any = null
  let items: TaxCreditsIncentives[] = tax_credit_incentives.filter((item) => {
    if (item.item === 'Electric Panel') {
      eletricPanel = item
    }
    return item.type === 'tax_credit' && item.item !== 'Electric Panel'
  })

  items = items?.filter(
    (item, index, self) =>
      index ===
      self.findIndex((t) => {
        return t.item === item.item
      })
  )

  if (items && eletricPanel) {
    eletricPanel.amount = 600
    eletricPanel['item'] = 'SPAN Smart Electrical Panel'
    items.unshift(eletricPanel)
    items = makeTaxCreditsTooltip(items)
  }

  return items
}
