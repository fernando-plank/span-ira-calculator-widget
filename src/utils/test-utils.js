var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
/* eslint-disable import/export */
import React from 'react';
import theme from '@styles/theme';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
jest.mock('../components/Icon', () => {
    return jest.fn(() => React.createElement("div", null, "mocked icon"));
});
const customRender = (ui, _a = {}) => {
    var renderOptions = __rest(_a, []);
    return render(React.createElement(ThemeProvider, { theme: theme }, ui), renderOptions);
};
export * from '@testing-library/react';
export { customRender as render };
