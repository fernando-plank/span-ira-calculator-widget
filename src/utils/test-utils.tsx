/* eslint-disable import/export */
import React, { ReactElement } from 'react'

import theme from '@styles/theme'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

type CustomRenderProps = Omit<RenderOptions, 'queries'>

jest.mock('../components/Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

const customRender = (
  ui: ReactElement,
  { ...renderOptions }: CustomRenderProps = {}
) => render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions)

export * from '@testing-library/react'
export { customRender as render }
