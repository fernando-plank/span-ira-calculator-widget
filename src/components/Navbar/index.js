import React from 'react';
import Icon from '@components/Icon';
import * as S from './styles';
import Menu from '@assets/img/menu.svg';
import ExplorePanel from '@assets/img/explore-panel.svg';
const Navbar = () => {
    return (React.createElement(S.Wrapper, null,
        React.createElement(S.MenuIconWrapper, { "data-testid": "navbar-menu" },
            React.createElement(Menu, { alt: "menu", "data-testid": "menu-button", width: 32, height: 20 })),
        React.createElement(S.NavbarLogo, null,
            React.createElement(Icon, { icon: "Logo" })),
        React.createElement(S.ExplorePanelWrapper, { onClick: () => window.open('https://www.span.io/panel', '_self') },
            React.createElement(ExplorePanel, { "data-testid": "explore-panel-button", alt: "explore-panel", width: 149, height: 24 }))));
};
export default Navbar;
