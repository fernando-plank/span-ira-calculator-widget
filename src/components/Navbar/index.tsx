import React from 'react'
import Icon from '@components/Icon'
import * as S from './styles'
import Menu from '@assets/img/menu.svg'
import ExplorePanel from '@assets/img/explore-panel.svg'

const Navbar = () => {
  return (
    <S.Wrapper>
      <S.MenuIconWrapper data-testid="navbar-menu">
        <Menu
          alt="menu"
          data-testid="menu-button"
          width={32}
          height={20}
        />
      </S.MenuIconWrapper>
      <S.NavbarLogo>
        <Icon icon="Logo" />
      </S.NavbarLogo>
      <S.ExplorePanelWrapper
        onClick={() => window.open('https://www.span.io/panel', '_self')}
      >
        <ExplorePanel
          data-testid="explore-panel-button"
          alt="explore-panel"
          width={149}
          height={24}
        />
      </S.ExplorePanelWrapper>
    </S.Wrapper>
  )
}

export default Navbar
