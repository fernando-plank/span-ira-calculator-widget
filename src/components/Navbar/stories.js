import React from 'react';
import Navbar from '.';
export default {
    title: 'Navbar',
    component: Navbar
};
export const Default = () => React.createElement(Navbar, null);
