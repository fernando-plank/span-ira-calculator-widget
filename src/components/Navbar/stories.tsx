import React from 'react'
import { Story, Meta } from '@storybook/react'

import Navbar from '.'

export default {
  title: 'Navbar',
  component: Navbar
} as Meta

export const Default: Story = () => <Navbar />
