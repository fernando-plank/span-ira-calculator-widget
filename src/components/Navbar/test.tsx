import { screen } from '@testing-library/react'
import { render } from 'utils/test-utils'

import Navbar from '.'

jest.mock('../Icon', () => {
  return jest.fn(() => <div>mocked icon</div>)
})

describe('<Navbar />', () => {
  it('should render the explore planel button', () => {
    const { container } = render(<Navbar />)

    expect(screen.getByTestId('explore-panel-button')).toHaveAttribute(
      'src',
      'img/explore-panel.svg'
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the menu button', () => {
    const { container } = render(<Navbar />)

    expect(screen.getByTestId('menu-button')).toHaveAttribute(
      'src',
      'img/menu.svg'
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
