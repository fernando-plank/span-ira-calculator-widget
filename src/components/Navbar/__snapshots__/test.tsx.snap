// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render the explore planel button 1`] = `
.c0 {
  width: 100%;
  height: 64px;
  background: #FAFAFA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  width: 80px;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c2 {
  heigth: 24px;
  margin-right: 20px;
  cursor: pointer;
}

<div
  class="c0"
>
  <div
    class="c1"
    data-testid="navbar-menu"
  >
    <img
      alt="menu"
      data-nimg="1"
      data-testid="menu-button"
      decoding="async"
      height="20"
      loading="lazy"
      src="img/menu.svg"
      style="color: transparent;"
      width="32"
    />
  </div>
  <div
    class=""
  >
    <div>
      mocked icon
    </div>
  </div>
  <div
    class="c2"
  >
    <img
      alt="explore-panel"
      data-nimg="1"
      data-testid="explore-panel-button"
      decoding="async"
      height="24"
      loading="lazy"
      src="img/explore-panel.svg"
      style="color: transparent;"
      width="149"
    />
  </div>
</div>
`;

exports[`<Navbar /> should render the menu button 1`] = `
.c0 {
  width: 100%;
  height: 64px;
  background: #FAFAFA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  width: 80px;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c2 {
  heigth: 24px;
  margin-right: 20px;
  cursor: pointer;
}

<div
  class="c0"
>
  <div
    class="c1"
    data-testid="navbar-menu"
  >
    <img
      alt="menu"
      data-nimg="1"
      data-testid="menu-button"
      decoding="async"
      height="20"
      loading="lazy"
      src="img/menu.svg"
      style="color: transparent;"
      width="32"
    />
  </div>
  <div
    class=""
  >
    <div>
      mocked icon
    </div>
  </div>
  <div
    class="c2"
  >
    <img
      alt="explore-panel"
      data-nimg="1"
      data-testid="explore-panel-button"
      decoding="async"
      height="24"
      loading="lazy"
      src="img/explore-panel.svg"
      style="color: transparent;"
      width="149"
    />
  </div>
</div>
`;
